wordpress-mobile/WordPress-Android
# WP Stories integration: add EmojiCompat library for stories emoji support

This PR builds on top of  #12031 

This PR implements the EmojiCompat library for support of emoji in the Stories library.

### What / Why / How
This PR implements `EmojiCompat` so to be able to have emojis on devices that may not have  support for them on their installed Fonts.

This PR uses the latest `EmojiCompat` library which is supposed to encapsulate the problems we'd otherwise need to tackle separately (downloading the fonts, then using them).

Resources:
- Official training (a bit outdated, things are not exactly as explained there as support library v28.0.0 has moved to jetpack `androidx` package now, but the core mechanism still is the same) https://developer.android.com/guide/topics/ui/look-and-feel/emoji-compat#using-widgets-with-appcompat
- Video https://www.youtube.com/watch?v=sYGKUtM2ga8 
- demo sample app https://github.com/googlesamples/android-EmojiCompat

Basically followed the sample app https://github.com/googlesamples/android-EmojiCompat and implemented the core parts here.

#### On downloadable fonts
For more information on how downloadable fonts work, read [here](https://github.com/googlesamples/android-DownloadableFonts) and [here](https://www.raywenderlich.com/213-custom-and-downloadable-fonts-on-android)  (see the part titled "Download fonts as XML resources") and then [this article by Google's font engineering lead](https://medium.com/google-design/the-android-developers-guide-to-better-typography-97e11bb0e261)
(tl;dr: it can all be done through XML for the platform to perform automatically on first app run, and Android Studio sets up everything for you, as done with the EmojiCompat library in particular).


Notes:
- the certificates used for `FontRequest` are borrowed from Google's EmojiCompat sample app, I haven't found anywhere in the docs that these should (or should not) be the ones to be used, but judging from the package names it looks like these certs should correspond to Google's, as this is how the call looks like:
<img width="393" alt="Screen Shot 2019-08-23 at 08 40 03" src="https://user-images.githubusercontent.com/6597771/63590222-a028eb00-c581-11e9-9733-20cc795e7e51.png">
Also, if you add a reference to EmojiCompat these are indeed the same ones that Android Studio adds to your project. 
- the requested Font is as per the sample app `Noto Color Emoji Compat`, the information on which can be found here: https://www.google.com/get/noto/help/emoji/



#### To test:
1. tap on the FAB button on the Me screen, then tap on Story option on the bottom sheet
2. capture a photo
3. tap on the stickers button, to the left of the text button
4. observe the bottom sheet with emoji appears
5. tap on one
6. observe it gets added


PR submission checklist:
